plugins/faucet/connector.go

```diff
@@ -0,0 +1,126 @@
+	"github.com/iotaledger/goshimmer/packages/core/ledger/utxo"
+	"github.com/iotaledger/goshimmer/packages/core/ledger/vm/devnetvm"
+	"github.com/iotaledger/goshimmer/packages/core/ledger/vm/devnetvm/indexer"
+	"github.com/iotaledger/goshimmer/packages/core/mana"
+	"github.com/iotaledger/goshimmer/packages/core/tangleold"
+	"github.com/iotaledger/goshimmer/plugins/blocklayer"
+	manaPlugin "github.com/iotaledger/goshimmer/plugins/blocklayer"
+	"github.com/iotaledger/hive.go/core/identity"
+	"github.com/iotaledger/hive.go/core/types/confirmation"
+	"github.com/mr-tron/base58"
+	"github.com/pkg/errors"
+)
+
+type FaucetConnector struct {
+	tangle  *tangleold.Tangle
+	indexer *indexer.Indexer
+}
+
+func NewConnector(t *tangleold.Tangle, indexer *indexer.Indexer) *FaucetConnector {
+	return &FaucetConnector{
+		tangle:  t,
+		indexer: indexer,
+	}
+}
+
+func (f *FaucetConnector) UnspentOutputs(addresses ...address.Address) (unspentOutputs wallet.OutputsByAddressAndOutputID, err error) {
+	unspentOutputs = make(map[address.Address]map[utxo.OutputID]*wallet.Output)
+
+	for _, addr := range addresses {
+		f.indexer.CachedAddressOutputMappings(addr.Address()).Consume(func(mapping *indexer.AddressOutputMapping) {
+			f.tangle.Ledger.Storage.CachedOutput(mapping.OutputID()).Consume(func(output utxo.Output) {
+				if typedOutput, ok := output.(devnetvm.Output); ok {
+					f.tangle.Ledger.Storage.CachedOutputMetadata(typedOutput.ID()).Consume(func(outputMetadata *ledger.OutputMetadata) {
+						if !outputMetadata.IsSpent() {
+							addr := address.Address{AddressBytes: typedOutput.Address().Array()}
+							walletOutput := &wallet.Output{
+								Address:                  addr,
+								Object:                   typedOutput,
+								ConfirmationStateReached: outputMetadata.ConfirmationState().IsAccepted(),
+								Spent:                    false,
+								Metadata: wallet.OutputMetadata{
+									Timestamp: outputMetadata.CreationTime(),
+								},
+							}
+
+							// store output in result
+							if _, addressExists := unspentOutputs[addr]; !addressExists {
+								unspentOutputs[addr] = make(map[utxo.OutputID]*wallet.Output)
+							}
+							unspentOutputs[addr][typedOutput.ID()] = walletOutput
+						}
+					})
+				}
+			})
+		})
+	}
+
+	return
+}
+
+func (f *FaucetConnector) SendTransaction(tx *devnetvm.Transaction) (err error) {
+	// attach to block layer
+	issueTransaction := func() (*tangleold.Block, error) {
+		block, e := deps.Tangle.IssuePayload(tx)
+		if e != nil {
+			return nil, e
+		}
+		return block, nil
+	}
+
+	_, err = blocklayer.AwaitBlockToBeBooked(issueTransaction, tx.ID(), maxTxBookedAwaitTime)
+	if err != nil {
+		return errors.Errorf("%v: tx %s", err, tx.ID().String())
+	}
+	return nil
+}
+
+func (f *FaucetConnector) RequestFaucetFunds(address address.Address, powTarget int) (err error) {
+	return nil
+}
+
+func (f *FaucetConnector) GetAllowedPledgeIDs() (pledgeIDMap map[mana.Type][]string, err error) {
```

Comment:
Not sure how this is internally used but it seems that there shouldn't be any allowed pledge IDs be configured on the faucet node and then the mana plugin will always just return the faucet nodeID in the allowed list. The pledges are still created correctly. So maybe this function can be simplified by just returning the current nodeID ?